### ISSUE TITLE
FN S1117: Add support for query comprehension range variables 

### ISSUE THREAD

**Issue by martin-strecker-sonarsource** [SonarSource/sonar-dotnet#6480]
### Description

S1117 (Local variables should not shadow class fields) should warn for range variables shadowing fields too.
```cs
    public void SomeMethod(byte[] byteArray)
    {
        _ = from a in byteArray
            let b = a.ToString()
            group b by b into g
            join d in new object[0] on g equals d
            select g into e
            select e;
    }
```
In this case the variables a, b, g, d, and e could shadow fields.

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1796](https://sonarsource.atlassian.net/browse/NET-1796)

[NET-1796]: https://sonarsource.atlassian.net/browse/NET-1796?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by pavel-mikula-sonarsource** [comment]
Moved
